src/Configuration/GuessedPaths.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GrumPHP\Configuration;
+
+use GrumPHP\Locator\GitDirLocator;
+use GrumPHP\Util\ComposerFile;
+use GrumPHP\Util\Filesystem;
+
+class GuessedPaths
+{
+    /**
+     * @var string
+     */
+    private $gitDir;
+
+    /**
+     * @var string
+     */
+    private $workingDir;
+
+    /**
+     * @var ComposerFile
+     */
+    private $composerFile;
+
+    /**
+     * @var string
+     */
+    private $binDir;
+
+    /**
+     * @var string
+     */
+    private $defaultConfigFile;
+
+    private function __construct()
+    {
+    }
+
+    public static function guess(Filesystem $filesystem, GitDirLocator $gitDirLocator): self
```

Comment:
@Landerstraeten : moved all path guessing logic to this factory. (probably gonna move the code though)
Can you go through it to see if it makes sense to you?

This one will be constructed early in the cli boot code and will be used to tell GrumPHP where to look for stuff. (git, bins, configs, ...)